WooCommerce/WooCommerceTests/ViewRelated/Products/Edit Product/ProductFormViewModel+ChangesTests.swift

```diff
@@ -271,4 +271,58 @@ final class ProductFormViewModel_ChangesTests: XCTestCase {
         // Assert
         XCTAssertTrue(viewModel.hasUnsavedChanges())
     }
+
+    func test_product_has_unsaved_changes_from_editing_downloadableFiles() {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+        let productImageActionHandler = ProductImageActionHandler(siteID: defaultSiteID, product: model)
+        let viewModel = ProductFormViewModel(product: model,
+                                             formType: .edit,
+                                             productImageActionHandler: productImageActionHandler,
+                                             isEditProductsRelease3Enabled: false,
+                                             isEditProductsRelease5Enabled: false)
+
+        // Action
+        viewModel.updateDownloadableFiles(downloadableFiles: MockProduct().sampleDownloadsMutated(), downloadLimit: 1, downloadExpiry: 1)
+
+        // Assert
+        XCTAssertTrue(viewModel.hasUnsavedChanges())
+    }
+
+    func test_product_has_unsaved_changes_from_editing_downloadLimit() {
+        // Arrange
+        let product = MockProduct().product(downloadable: true)
+        let model = EditableProductModel(product: product)
+        let productImageActionHandler = ProductImageActionHandler(siteID: defaultSiteID, product: model)
+        let viewModel = ProductFormViewModel(product: model,
+                                             formType: .edit,
+                                             productImageActionHandler: productImageActionHandler,
+                                             isEditProductsRelease3Enabled: false,
+                                             isEditProductsRelease5Enabled: false)
+
+        // Action
+        viewModel.updateDownloadableFiles(downloadableFiles: MockProduct().sampleDownloads(), downloadLimit: 5, downloadExpiry: 1)
```

Comment:
nit: you shouldn't need to edit the `downloadExpire` here.

Reply:
I am not actually editing `downloadExpire ` I am only editing `downloadLimit`.  Because in. `MockProduct().product(downloadable: true)`, the value for both `downloadExpire` and `downloadLimit` are `1`. SO at this line I am only changing `downloadLimit` to `5`. And since both value are non optional so, I must pass the original value here which is `1`